app/scripts/metamask-controller.js

```diff
@@ -852,8 +855,83 @@ export default class MetamaskController extends EventEmitter {
             saveState: async () => {
               await this.keyringController.persistAllKeyrings();
             },
-            removeAccount: async (address) => {
-              await this.removeAccount(address);
+            addAccount: async (_address, origin, handleUserInput) => {
+              const { id: addAccountApprovalId } =
+                this.approvalController.startFlow();
+
+              const confirmationResult =
+                await this.approvalController.addAndShowApprovalRequest({
+                  origin,
+                  type: SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES.confirmAccountCreation,
+                  expectsResult: true,
+                });
+
+              if (confirmationResult.value) {
+                try {
+                  await handleUserInput(confirmationResult.value);
+                  await this.keyringController.persistAllKeyrings();
+                  await this.approvalController.success({
+                    flowToEnd: addAccountApprovalId,
+                    message: 'snapAccountCreated', // translation key for message
```

Comment:
We shouldn't need to use translation keys here as the extension background now supports translation also using the `t` function from `app/scripts/translate.ts`. It's exactly the same API as the frontend.

Reply:
Ah this is perfect! Thank you